_rules/image-button-accessible-name-59796f.md

```diff
@@ -55,128 +55,105 @@ There is a known combination of a popular browser and assistive technology that
+<input type="image" src="/test-assets/shared/search-icon.png" title="Search" />
 ```
 
 #### Passed Example 4
 
-Image button element with [accessible name][] through `aria-labelledby`
+The image button has an [accessible name][] through the `aria-labelledby` attribute.
 
 ```html
-<input type="image" name="submit" src="button.gif" aria-labelledby="id1" />
-<div id="id1">Submit</div>
-```
-
-#### Passed Example 5
-
-[accessible name][] is not empty.
-
-```html
-<input type="image" name="submit" src="button.gif" alt=":-)" />
-```
-
-#### Passed Example 6
-
-Image button element with [accessible name][] through `alt` attribute
-
-```html
-<input type="image" name="submit" src="button.gif" alt="123" />
+<input type="image" src="/test-assets/shared/search-icon.png" aria-labelledby="id1" />
+<div id="id1">Search</div>
 ```
 
 ### Failed
 
 #### Failed Example 1
 
-Image button element with empty [accessible name][].
+The image button element has an empty [accessible name][]. The `name` attribute can not be used to provide an [accessible name][].
 
 ```html
-<input type="image" name="submit" src="button.gif" />
+<input type="image" name="search" src="/test-assets/shared/search-icon.png" />
 ```
 
 #### Failed Example 2
 
-Image button element with empty [accessible name][].
+The image button has an empty `alt` attribute, and no other attributes that can give it an [accessible name][].
 
 ```html
-<input type="image" name="submit" src="button.gif" alt="" />
+<input type="image" src="/test-assets/shared/search-icon.png" alt="" />
 ```
 
 #### Failed Example 3
 
-Image button with empty [accessible name][] because the `aria-labelledby` attribute does not reference an id that exists in the same document.
-
-```html
-<input type="image" name="submit" src="button.gif" aria-labelledby="id1" />
-```
-
-#### Failed Example 4
-
-[accessible name][] is empty.
+The image button has an `aria-labelledby` attribute, but the referenced element does not exist. This gives the button an empty [accessible name][].
 
 ```html
-<input type="image" name="submit" src="button.gif" alt=" " />
+<input type="image" src="/test-assets/shared/search-icon.png" aria-labelledby="non-existing" />
 ```
 
 ### Inapplicable
 
 #### Inapplicable Example 1
 
-HTML `button` element is not an image button
+The `button` element is not an image button. [Success Criterion 1.1.1 Non-text Content](https://www.w3.org/TR/WCAG21/#non-text-content) can not fail text buttons. Only non-text content is applicable.
 
 ```html
 <button>My button</button>
 ```
 
 #### Inapplicable Example 2
 
-HTML `input` element with type with a `type` attribute in the `Button` state is not an image button
+The `input` element with type with a `type` attribute in the `Button` state is not an image button. [Success Criterion 1.1.1 Non-text Content](https://www.w3.org/TR/WCAG21/#non-text-content) can not fail text buttons. Only non-text content is applicable.
```

Comment:
```suggestion
The `input` element with a `Button` value in the `type` attribute is not an image button. [Success Criterion 1.1.1 Non-text Content](https://www.w3.org/TR/WCAG21/#non-text-content) can not fail text buttons. Only non-text content is applicable.
```

Just suggesting, since I guess this is the message we want to communicate here. `Button` is a value for the `type` attribute. I am not sure what "state" tries to convey there.

Reply:
@daniel-montalvo: the `type` attribute (and a few other) can have any string as value. These values are converted to "states" which are only from a limited (finite) set.

Notably, both the "Button" and "button" **values** will be converted into the "Button" **state** (ie, case insensitivity).

So, it is indeed the **state** of the attribute we want to look at…
https://html.spec.whatwg.org/multipage/input.html#attr-input-type